File: hw/collahuazo/u3/hw24CompositePaterns/CompositePatternPy/composite/model/president.py
from composite.model.supervisor import Supervisor


class President(Supervisor):

    _president = None 

    def __init__(self):
        super().__init__()
        self.title = "President"

    def state_name(self):
        super().state_name()

    @classmethod
    def get_president(cls, a_name):
        if cls._president is None:
            cls._president = President()
        cls._president.name = a_name
        return cls._president

File: hw/aimacana/u3/hw25CompositePattern/java/Composite/src/main/java/ec/edu/espe/composite/model/President.java
package ec.edu.espe.composite.model;

public class President extends Supervisor {
    private static President president = new President();
    
    private President() {
        super();
        title = "President";
    }

    private President(String aName) {
        this();
        name = aName;
    }
    
    @Override
    public String stateName() {
        return super.stateName();
    }
    
    public static President getPresident(String aName) {
        president.name = aName;
        return President.president;
    }
}
